src/orion/executor/multiprocess_backend.py

```diff
@@ -167,13 +167,18 @@ def __init__(self, n_workers=-1, backend="multiprocess", **kwargs):
         if n_workers <= 0:
             n_workers = multiprocessing.cpu_count()
 
+        self.pool_config = {"n_workers": n_workers, "backend": backend}
         self.pool = PoolExecutor.BACKENDS.get(backend, ThreadPool)(n_workers)
 
     def __setstate__(self, state):
-        self.pool = state["pool"]
+        log.warning("Nesting multiprocess executor")
```

Comment:
So the workaround is to pass the config instead of the pool and recreate a pool with same config inside of it?

Reply:
Unless I am getting something wrong, it seems to me we are increasing the number of workers exponentially by handling nested submits this way. Say I pass the executor to a function to be executed by submit, then `n` workers would be created inside this subprocess, in addition to the already existing `n` workers.

Reply:
No the workaround is `return _Process(*args, **kwds)`, `_Process` being a process that cannot be a daemon.
The daemon check is done inside the worker code by an assert, to bypass the assert the worker process should not be a daemon.